doc/source/whatsnew/v0.25.0.rst

```diff
@@ -741,6 +741,47 @@ consistent with NumPy and the rest of pandas (:issue:`21801`).
    cat.argsort()
    cat[cat.argsort()]
 
+.. _whatsnew_0250.api_breaking.list_of_dict:
+
+Column order is preserved when passing a list of dicts to DataFrame
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Starting with Python 3.7 the key-order of ``dict`` is `guaranteed <https://mail.python.org/pipermail/python-dev/2017-December/151283.html>`_. In practice, this has been true since
+Python 3.6. The :class:`DataFrame`  constructor now treats a list of dicts in the same way as
+it has a list of ``OrderedDict`` since v0.19.0. This change applies only when pandas is running
+on Python>=3.6. As a consequence, the column order produced by :class:`DataFrame`
```

Comment:
This text is awkward and confusing starting with 'This change applies only when pandas is running...' pls simplify.

Reply:
Joris kindly rewrote that part. If that's not good enough for you either, please suggest a specific revision.